barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/ultra_flavor.hpp

```diff
@@ -47,10 +48,21 @@ class UltraFlavor {
     static constexpr size_t NUM_PRECOMPUTED_ENTITIES = 27;
     // The total number of witness entities not including shifts.
     static constexpr size_t NUM_WITNESS_ENTITIES = 8;
-    // The total number of witnesses including shifts and derived entities.
-    static constexpr size_t NUM_ALL_WITNESS_ENTITIES = 13;
     // Total number of folded polynomials, which is just all polynomials except the shifts
     static constexpr size_t NUM_FOLDED_ENTITIES = NUM_PRECOMPUTED_ENTITIES + NUM_WITNESS_ENTITIES;
+    // The index of the shift of the first unshifted witness
+    static constexpr size_t NUM_SHIFTED_WITNESSES = 5;
+
+    static constexpr size_t NUM_SHIFTED_TABLES = 4;
+
+    // A container to be fed to ShpleminiVerifier to avoid redundant scalar muls
+    static constexpr RepeatedCommitmentsData REPEATED_COMMITMENTS =
```

Comment:
still somewhat ugly here, but I think expressing everythin in terms of numbers of entities is a better practice than just constants with the required indexes. e.g. someone would delete shifted tables - delete the constant - nothing would be broken